### ISSUE TITLE
BoundingBox3D description unclear

### ISSUE THREAD

**Issue by ryan-roche** [Kukanani/vision_msgs_proposal#101]
I'm having trouble understanding how exactly to use the fields within the BoundingBox3D message.

What direction relative to the box should the orientation pose be pointing? Should it be along one specific axis, and the rest are relative to it by the right-hand rule?

What do the X, Y, and Z values of the size field represent? Are they the full length/height/width values measured face-to-face on the box, or are they the half-values (the distance between the center and the corresponding faces)?

I suggest editing the comments and official documentation to clarify some standard for what those values represent to ensure that different projects using this message type are entering the correct measurements.

**Comment by mintar** [comment]
> What direction relative to the box should the orientation pose be pointing?

The pose specifies the pose of the bounding box in the `header.frame_id` frame. 

> What do the X, Y, and Z values of the size field represent?

The size represents the full size of the bbox (face to face). The `x` value is the size of the bbox along the `x` axis of the local coordinate frame of the bbox as defined by the `pose` and so on for the other axes.

I hope that clarifies things. If not, tell me and I'll try to draw it tomorrow.

**Comment by ryan-roche** [comment]
I think I understand. The x/y/z values of the size represent the side lengths along the corresponding axes in the LOCAL coordinate frame of the bounding box that is obtained by applying the quaternion transformation to the header.frame_id origin axes

**Comment by mintar** [comment]
Yes, exactly! Not just the quaternion rotation of course, but the full transform including the translation. But I guess that's what you meant.

**Comment by ryan-roche** [comment]
Just to confirm, the positive X direction of the bounding box's local coordinate system is the direction the pose "points" to?

Like this?
<img width="467" alt="image" src="https://github.com/ros-perception/vision_msgs/assets/48164514/790e0a72-5daf-423f-be37-2684c5efd2b8">

**Comment by mintar** [comment]
Yes, you could say that. Although I think it's not good to think of a pose as an arrow like in your image, because you lose the information how the pose is rotated around the x axis. It's better to think of a pose as a local coordinate system with its own x, y and z axes, like this:

![grafik](https://github.com/ros-perception/vision_msgs/assets/320188/18eb17bb-3726-472f-8184-60e742c3b846)

Some tools (like RViz) allow you to switch the pose visualization mode between "axes" and "arrow", and by convention the arrow always points into the direction of the x axis, so your image is not wrong.

**Comment by ryan-roche** [comment]
I was about to say, isn't the convention that positive-X is the "forwards" direction? That's what I meant by the green arrow on the x-axis of the pose coordinate system.

**Comment by mintar** [comment]
Yes, that's exactly right. All I was saying is that an arrow is ambiguous (it shows the "forwards" direction, but not the "up" direction), not that it's wrong!
